### PR TITLE
📖  add v1.2 -> v1.3 migration doc

### PR DESCRIPTION
**Pull request by ykakarap**
**What this PR does / why we need it**:

This PRs adds `v1.2` => `v1.3` migration doc.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

Part of #6615

### CODE REVIEW

File: docs/book/src/developer/providers/v1.2-to-v1.3.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Cluster API v1.2 compared to v1.3
+
+This document provides an overview over relevant changes between ClusterAPI v1.2 and v1.3 for
+maintainers of providers and consumers of our Go API.
+
+## Minimum Go version
+
+* The Go version used by Cluster API is now Go 1.17.x
```
Comment (sbueringer) [inline docs/book/src/developer/providers/v1.2-to-v1.3.md:8]:
```suggestion
* The Go version used by Cluster API is now Go 1.18.x
```
nit

File: docs/book/src/developer/providers/v1.2-to-v1.3.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Cluster API v1.2 compared to v1.3
+
+This document provides an overview over relevant changes between ClusterAPI v1.2 and v1.3 for
```
Comment (killianmuldoon) [inline docs/book/src/developer/providers/v1.2-to-v1.3.md:3]:
```suggestion
This document provides an overview over relevant changes between Cluster API v1.2 and v1.3 for
```

**Review by killianmuldoon — COMMENTED** [review]
One tiny nit -  otherwise 

/lgtm

File: docs/book/src/developer/providers/v1.2-to-v1.3.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+maintainers of providers and consumers of our Go API.
+
+## Minimum Go version
+
+* The Go version used by Cluster API is now Go 1.17.x
+
+## Dependencies
+
+**Note**: Only the most relevant dependencies are listed, `k8s.io/` and `ginkgo`/`gomega` dependencies
+in ClusterAPI are kept in sync with the versions used by `sigs.k8s.io/controller-runtime`.
```
Comment (oscr) [inline docs/book/src/developer/providers/v1.2-to-v1.3.md:13]:
```suggestion
in Cluster API are kept in sync with the versions used by `sigs.k8s.io/controller-runtime`. 
```

**Comment by sbueringer** [comment]
lgtm pending squash

**Comment by ykakarap** [comment]
squashed

**Comment by ykakarap** [comment]
> /lgtm /approve /cherry-pick release-1.2

We dont need this in the release-1.2 brach right? This doc is taking about migration from v1.2 to v1.3.

The docs we will need in release-1.2 are migration to v1.2 not from v1.2, correct?

**Comment by sbueringer** [comment]
Ups

**Comment by sbueringer** [comment]
Can you please open a revert PR? 😃

**Comment by ykakarap** [comment]
> Can you please open a revert PR? 😃

will do.

**Comment by ykakarap** [comment]
Revert PR: https://github.com/kubernetes-sigs/cluster-api/pull/6730

**Pull request merged**